packages/analytics/amplify_analytics_pinpoint/example/integration_test/events_test.dart

```diff
@@ -135,11 +97,129 @@ void main() {
               // There is some lossiness behavior to doubles
               Map.fromEntries([lossyDoubleProperty, intProperty]),
             ),
+            // Session is valid before timestamp
+            containsPair(
+                'endpoint',
+                allOf(
+                  containsPair('EndpointStatus', 'ACTIVE'),
+                  containsPair('OptOut', 'ALL'),
+                  containsPair(
+                    'Location',
+                    allOf(
+                      containsPair('Latitude', 0.0),
+                      containsPair('Longitude', 0.0),
+                    ),
+                  ),
+                  containsPair('Attributes', {}),
+                  containsPair('Metrics', {}),
+                ))
+          ])),
+        );
+
+        await Amplify.Analytics.recordEvent(event: customEvent);
+        await Amplify.Analytics.flushEvents();
+      },
+      timeout: const Timeout(Duration(minutes: 2)),
+    );
+
+    test(
+      'Analytics.register and unregister of GlobalProperties adds and removes properties in future events',
+      () async {
+        const customEventName = 'my custom event type name';
+        final customEvent = AnalyticsEvent(customEventName);
+
+        // Add local event properties
+        customEvent.properties.addStringProperty(
+          secondStringProperty.key,
+          secondStringProperty.value,
+        );
+        customEvent.properties.addIntProperty(
+          secondIntProperty.key,
+          secondIntProperty.value,
+        );
+
+        // Add attribute global property types
+        final attributeGlobalProperties = AnalyticsProperties()
+          ..addBoolProperty(boolProperty.key, boolProperty.value)
+          ..addStringProperty(stringProperty.key, stringProperty.value);
+
+        await Amplify.Analytics.registerGlobalProperties(
```

Comment:
Will global properties persist during tests? If so, should you be calling `Amplify.reset` after each test or should this go in its own group?

Reply:
in this test, I ensure no global properties are persisted by calling: 
        await Amplify.Analytics.unregisterGlobalProperties();

Which unregisters all global properties.